aesara/tensor/extra_ops.py

```diff
@@ -1534,13 +1534,19 @@ def broadcast_shape_iter(
                 result_dims.append(maybe_non_bcast_shapes[0])
                 continue
 
-            non_bcast_vec = at.as_tensor(maybe_non_bcast_shapes)
-            non_bcast_vec = at.switch(eq(non_bcast_vec, 1), -one_at, non_bcast_vec)
-            dim_max = at_abs(at_max(non_bcast_vec))
+            non_bcast_vec = [
```

Comment:
Why didn't the elemwise version work?

Reply:
There were some tests that checked to make sure no `Elemwise` `Op`s were present in their graphs, and they mostly made sense.

More importantly, there's an unnecessary sort of circular dependency when `Elemwise.infer_shape` produces `Elemwise`s.  The things we're iterating over are necessarily `tuple`s, so introducing `Elemwise` is a bit of overkill.

The only thing I don't particularly like about this scalar `Op` approach is that the graphs it produces are less likely to be covered by rewrites, because most of our rewrites are implemented for `Elemwise` `Op`s and not their corresponding scalar `Op`s.  Regardless, this is a more _general_ issue that should be addressed separately, if/when need be&mdash;and there doesn't appear to be a need right now (via the tests).